### PR TITLE
#162: Belos: Provide `Tpetra` version of RCG example

### LINKED ISSUE #162: Belos: Provide `Tpetra` version of RCG example

### PR DESCRIPTION
**Pull request by antoinemeyer5**
Fixes #162

### CODE REVIEW

File: packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+  return ( success ? EXIT_SUCCESS : EXIT_FAILURE );
+}
+
+int main(int argc, char *argv[]) {
+  run<double>(argc,argv);
+
+  // wrapped with a check: CMake option Trilinos_ENABLE_FLOAT=ON
+  // run<float>(argc,argv);
+}
```
Comment (tlamonthezie) [inline packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp]: Hello @antoinemeyer5 I find there some issue:
The main function should return the result of the run call. Could you please add a `return` statement ?

Comment (antoinemeyer5) [reply packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp]: Fix, thanks!

File: packages/belos/tpetra/example/RCG/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_TpetraExt)
+ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_Triutils)
```
Comment (tlamonthezie) [inline packages/belos/tpetra/example/RCG/CMakeLists.txt]: Hello @antoinemeyer5 is TriUtils  required for this test ?
I think it is only required when using BelosTpetraTestFramework or BelosTpetraUtils

File: packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    }
+    
+    // Perform solve
+    Belos::ReturnType ret = newSolver->solve();
+    
+    // Compute actual residuals
+    bool badRes = false;
+    std::vector<ST> actual_resids( numrhs );
+    std::vector<ST> rhs_norm( numrhs );
+    MV resid(rowMap, numrhs);
```
Comment (tlamonthezie) [inline packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp:191]: I am not sure that using snaked case variable names here is consistent with the rest of the file

File: packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+#include <Tpetra_MatrixIO.hpp>
+
+// Belos
+#include "BelosConfigDefs.hpp"
+#include "BelosLinearProblem.hpp"
+#include "BelosTpetraAdapter.hpp"
+#include "BelosRCGSolMgr.hpp"
+
+// Trilinos
+#include <Trilinos_Util.h>
```
Comment (tlamonthezie) [inline packages/belos/tpetra/example/RCG/RCGTpetraExFile.cpp]: Same remark as before. Is TrilUtils used in this file ?

**Review by tlamonthezie — COMMENTED** [review]
Sounds good

**Review by stmcgovern — APPROVED** [review]
LGTM

**Pull request merged**